### ISSUE TITLE
Not working after changing the spaCy language

### ISSUE THREAD

**Issue by potterthecop** [jackboyla/GLiREL#15]
Hello, guys.

I'm trying to use this wonderful functionality but with Romanian language. So I have downloaded ro_core_news_lg from Spacy and modified the code structure.
But the relations aren't detected if I provide a Romanian text.

Can you help me regarding what am I doing wrong? I have to change another piece of code?

Thanks!

**Comment by jackboyla** [comment]
Hey @potterthecop , awesome to see you trying to make GLiREL work for Romanian! I quickly tried it myself using the code below and it seems to work. I'm not sure about the quality as sadly I don't know Romanian, and the model likely hasn't seen much Romanian during training🥲 Let me know if this works for you!

```python
from glirel import GLiREL
import spacy

model = GLiREL.from_pretrained("jackboyla/glirel-large-v0")

nlp = spacy.load('ro_core_news_sm')

text = """
București, 31 ianuarie 2025 – Economia României a înregistrat o creștere surprinzătoare în ultimul trimestru al anului 2024, depășind previziunile analiștilor economici.
"""
doc = nlp(text)
tokens = [token.text for token in doc]

labels = ['country of origin', 'licensed to broadcast to', 'father', 'followed by', 'characters']

ner = [[ent.start, ent.end-1, ent.label_, ent.text] for ent in doc.ents]

print(f"NER -> {ner}")

relations = model.predict_relations(tokens, labels, threshold=0.0, ner=ner, top_k=1)

print('Number of relations:', len(relations))

sorted_data_desc = sorted(relations, key=lambda x: x['score'], reverse=True)
print("\nDescending Order by Score:")
for item in sorted_data_desc:
    print(f"{item['head_text']} --> {item['label']} --> {item['tail_text']} | score: {item['score']}")
```


```
NER -> [[1, 1, 'GPE', 'București'], [3, 5, 'DATETIME', '31 ianuarie 2025'], [7, 8, 'ORGANIZATION', 'Economia României'], [15, 19, 'DATETIME', 'ultimul trimestru al anului 2024']]
Number of relations: 12

Descending Order by Score:
['București'] --> country of origin --> ['Economia', 'României'] | score: 0.48268017172813416
['Economia', 'României'] --> country of origin --> ['București'] | score: 0.3390689492225647
['București'] --> country of origin --> ['31', 'ianuarie', '2025'] | score: 0.08241847157478333
['31', 'ianuarie', '2025'] --> country of origin --> ['Economia', 'României'] | score: 0.06979944556951523
['ultimul', 'trimestru', 'al', 'anului', '2024'] --> country of origin --> ['București'] | score: 0.06848997622728348
['31', 'ianuarie', '2025'] --> country of origin --> ['București'] | score: 0.06690288335084915
['București'] --> country of origin --> ['ultimul', 'trimestru', 'al', 'anului', '2024'] | score: 0.05789363384246826
['Economia', 'României'] --> country of origin --> ['31', 'ianuarie', '2025'] | score: 0.03781758248806
['ultimul', 'trimestru', 'al', 'anului', '2024'] --> country of origin --> ['Economia', 'României'] | score: 0.03145815059542656
['Economia', 'României'] --> country of origin --> ['ultimul', 'trimestru', 'al', 'anului', '2024'] | score: 0.016709787771105766
['31', 'ianuarie', '2025'] --> followed by --> ['ultimul', 'trimestru', 'al', 'anului', '2024'] | score: 8.959906949712604e-07
['ultimul', 'trimestru', 'al', 'anului', '2024'] --> followed by --> ['31', 'ianuarie', '2025'] | score: 5.39177790415124e-07
```

**Comment by chrishokamp** [comment]
@potterthecop if you want to try to fine tune, you could follow @jackboyla's example here https://github.com/jackboyla/GLiREL/blob/main/examples/finetune.ipynb 
and repeat the annotation protocol for romanian using an llm like here https://huggingface.co/utter-project/EuroLLM-9B-Instruct 
please do continue the thread if you want to do this and you need more guidance

**Comment by potterthecop** [comment]
> Hey [@potterthecop](https://github.com/potterthecop) , awesome to see you trying to make GLiREL work for Romanian! I quickly tried it myself using the code below and it seems to work. I'm not sure about the quality as sadly I don't know Romanian, and the model likely hasn't seen much Romanian during training🥲 Let me know if this works for you!
> 

Awesome, this code works for me, also. I have used the second code posted on the front page and it didn't work.
Thanks so much!

**Comment by potterthecop** [comment]
> [@potterthecop](https://github.com/potterthecop) if you want to try to fine tune, you could follow [@jackboyla](https://github.com/jackboyla)'s example here https://github.com/jackboyla/GLiREL/blob/main/examples/finetune.ipynb and repeat the annotation protocol for romanian using an llm like here https://huggingface.co/utter-project/EuroLLM-9B-Instruct please do continue the thread if you want to do this and you need more guidance

Hi, @chrishokamp ,
I'd love to do some fine tunning because, working in law enforcement, we have some specific terms when we fill some papers and I feel the need to train the used model.
I have used EuroLLM-9B-Instruct in a software named Knime in order to detect entities and relationships from a given text, it was ok but that's it. I'm a real noob regarding fine tunning ops so..if you have some spare time to detail the steps that I have to take, based on @jackboyla 's example, it would be great.
Thanks alot!

**Comment by chrishokamp** [comment]
@potterthecop so what I would try first is follow this script -- https://github.com/jackboyla/GLiREL/blob/main/data/dataset-generation/2_vllm_data_gen.py#L133-L135 

but augment the prompt to _also_ translate to Romanian -- this will result in a Romanian training dataset for glirel
if you are working in a specific domain you might also try to augment the dataset generation to discard examples that are not in the domain

Once you have a training dataset, i would try training a targeted romanian model first, following from here https://github.com/jackboyla/GLiREL/blob/main/train.py#L539

> I'm a real noob regarding fine tunning ops so..if you have some spare time to detail the steps that I have to take, based on @jackboyla 's example, it would be great.

feel free to create a working PR to this repo as you go and we may be able to advise on specifics

>Another thing regarding the labels. The labels have to be created exactly how you encounter them in the text or they have to be synonyms of entity relationships or ... you have to follow some rules?

No, the point of zero shot is that you don't need to know the labels upfront, but your training data should at least contain data that's similar to the domain you want to predict on
